### PR TITLE
enable staticcheck [CHA-3542]

### PR DESCRIPTION
**Pull request by rooftoofwoof**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -20,6 +20,14 @@ jobs:
     - name: Build
       run: GOWORK=off CGO_ENABLED=0 go build -v
 
+    - name: staticcheck
```
Comment (ravirahman) [inline .github/workflows/test.yml:23]: Can we enable staticheck for all go projects (bigquery-streaming-write-loader, engine proxy, ...)

File: internal/feather.go [hunk 2]
```diff
@@ -524,30 +530,36 @@ func ChalkMarshal(attrs map[string]any) ([]byte, error) {
 	// Magic string header
 	_, err := ioWriter.WriteString("CHALK_BYTE_TRANSMISSION")
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "failed to write chalk transmission magic string")
```
Comment (ravirahman) [inline internal/feather.go:533]: nit: I see both fmt.Errorf and errors.Wrap

File: .github/workflows/test.yml (hunk 1 above)
Comment (rooftoofwoof) [reply .github/workflows/test.yml:23]: it shall arrive one day

**Review by ravirahman — APPROVED** [review]

File: internal/feather.go (hunk 2 above)
Comment (rooftoofwoof) [reply internal/feather.go:533]: thank you

**Pull request merged**